File: apps/ayokoding-www/content/en/learn/fundamentally-strong/software-engineer/just-enough-python/learning/code/ex-05-lint-with-ruff/bad.py
"""Example 5: Lint With Ruff -- deliberately has an unused import."""

# Never referenced below -- ruff's F401 rule flags exactly this.
import json  # => imports the json module but never uses it (deliberately unused)


def greet(name: str) -> str:  # => defines greet, takes name, returns a str
    return f"hello {name}"  # => builds and returns "hello Ada" when called below


print(greet("Ada"))  # => calls greet("Ada"), prints "hello Ada"
# => Output: hello Ada -- the SCRIPT runs fine; ruff still flags it

File: apps/ayokoding-www/content/en/learn/courses/just-enough-typescript/learning/code/ex-77-eslint-clean/bad.ts
// Example 77: eslint Clean -- bad.ts has a genuine unused variable.
function greet(name: string): string {
  const unused = "never read"; // => eslint's no-unused-vars rule flags exactly this
  return `hi ${name}`; // => the function itself still works correctly
}

console.log(greet("Ada")); // => Output: hi Ada -- the SCRIPT runs fine; eslint still flags it
